src/sqlfluff/rules/L064.py

```diff
@@ -0,0 +1,260 @@
+"""Implementation of Rule L064."""
+
+from typing import Optional
+
+import regex
+
+from sqlfluff.core.parser.segments.raw import CodeSegment
+from sqlfluff.core.rules.base import BaseRule, LintFix, LintResult, RuleContext
+from sqlfluff.core.rules.doc_decorators import (
+    document_configuration,
+    document_fix_compatible,
+)
+
+
+@document_configuration
+@document_fix_compatible
+class Rule_L064(BaseRule):
+    r"""Consistent usage of preferred quotes for quoted literals.
+
+    Some databases allow single quotes as well as double quotes to express a quoted
+    literal. Prefer one type of quotes as specified in rule setting. Falling back to
+    alternate quotes to reduce escapes.
+
+    Dollar quoted raw strings are excluded from this rule, as they are mostly used for
+    literal UDF Body definitions.
+
+    .. note::
+       This rule only checks quoted literals and not quoted identifiers as they often
+       cannot interchange single and double quotes
+
+       This rule can be dangerous. If accidentally enabled for dialects that do not
```

Comment:
Suggest we don’t need the scary first sentence since we’re working on inclusion list:

```suggestion
       If accidentally enabled for dialects that do not
```

Btw might want to change line breaks rather than just accept this.